src/NUnitCore/core/CoreExtensions.cs

```diff
@@ -221,12 +221,22 @@ public void InstallAddins()
 			}
 		}
 
+        public ArrayList AdhocExtensions = new ArrayList();
```

Comment:
Could this be `public IList<object> AdhocExtensions  { get; } = new List<object>();` so that callers aren't bound to the concete type and so that there aren't two ways to mutate (setting the field and mutating the instance in the field)?

Reply:
Actually, why not `IList<Addin>`?

Reply:
That's how I would write it in today's world, obviously, and I considered it here. But we use `ArrayList` everywhere else because this code actually compiled and was released for .NET 1.x up through (I think) 2.6.3. If I changed it here, I'd feel like I should change it in a bunch of other places and also change all the other `ArrayList`s to typed `List`s. So I conformed to what was already in the code.

Again... if this were to go beyond a point release to a 2.7, I'd do more. Pros and cons of that move for a later discussion.

Reply:
I see. Can you still make that a read-only property so that the ArrayList instance itself cannot be replaced outside this class?

Reply:
Sure. Actually, I got lazy and made it a field because the initialization syntax for properties doesn't work in this version of C# either. But it can be `{ get; private set; } ` with initialization in the constructor.

Reply:
You brought up a good point about makign things internal. `internal readonly` makes sense to me too.